packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4497,66 +4497,452 @@ class PositionedDirectional extends StatelessWidget {
   }
 }
 
-/// A widget that displays its children in a one-dimensional array.
-///
-/// The [Flex] widget allows you to control the axis along which the children are
-/// placed (horizontal or vertical). This is referred to as the _main axis_. If
-/// you know the main axis in advance, then consider using a [Row] (if it's
-/// horizontal) or [Column] (if it's vertical) instead, because that will be less
-/// verbose.
+/// A widget that is the basis for [Row] and [Column]. It arranges its
+/// children in a single line – either horizontally (a [Row]) or vertically
+/// (a [Column]).
+
+/// If you only have one child, consider using [Align] or [Center] to
+/// position it.
+
+/// **Key Features:**
+
+/// * **[direction]:**  Determines whether children are placed
+/// side-by-side ([Axis.horizontal]) or one above the other
+/// ([Axis.vertical]). Instead of using a [Flex] directly with this parameter,
+/// it is recommended to use [Row] and [Column].
 ///
-/// To cause a child to expand to fill the available space in the [direction]
-/// of this widget's main axis, wrap the child in an [Expanded] widget.
+/// Example usage: "direction: Axis.vertical".
 ///
-/// The [Flex] widget does not scroll (and in general it is considered an error
-/// to have more children in a [Flex] than will fit in the available room). If
-/// you have some widgets and want them to be able to scroll if there is
-/// insufficient room, consider using a [ListView].
+/// * **Layout:** By default, the [Flex] widget won't let its content scroll
+/// or wrap to the next line. If you try to squeeze in too many children,
+/// you'll see an "overflow error" (during development black and yellow
+/// stripes, but a production app just cuts off the content). For content
+/// that should wrap to a new line, use [Wrap]. For scrolling content
+/// prefer [SliverList] over [ListView] or [SingleChildScrollView], for
+/// better performance.
 ///
-/// The [Flex] widget does not allow its children to wrap across multiple
-/// horizontal or vertical runs. For a widget that allows its children to wrap,
-/// consider using the [Wrap] widget instead of [Flex].
+/// * **Performance:** It is recommended to check your app's
```

Comment:
The point about a [Wrap] was already in there, plus I think it's important to point this out to a new user who otherwise would have no idea that a [Wrap] is even an option.